From 707500d0f33f654c4a1de78af747cd126398144b Mon Sep 17 00:00:00 2001
From: Lukas Bulwahn <lukas.bulwahn@gmail.com>
Date: Wed, 19 Jun 2019 22:42:40 +0200
Subject: [PATCH 112/171] Revert "tty_io: failed experiment to make clang
 handle some cases"

This reverts commit b9ad7160ec245c064be1058c0c4929eec3c9c474.

The experiment failed so revert back.
---
 drivers/tty/tty_io.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/tty/tty_io.c b/drivers/tty/tty_io.c
index 2507261dbfad..b113d64e89e5 100644
--- a/drivers/tty/tty_io.c
+++ b/drivers/tty/tty_io.c
@@ -908,7 +908,7 @@ static inline ssize_t do_tty_write(
 	unsigned int chunk;
 
 	ret = tty_write_lock(tty, file->f_flags & O_NDELAY);
-	if (!ret)
+	if (ret < 0)
 		return ret;
 
 	/*
@@ -1085,7 +1085,7 @@ int tty_send_xchar(struct tty_struct *tty, char ch)
 		return 0;
 	}
 
-	if (!tty_write_lock(tty, 0))
+	if (tty_write_lock(tty, 0) < 0)
 		return -ERESTARTSYS;
 
 	down_read(&tty->termios_rwsem);
@@ -2382,7 +2382,7 @@ static int send_break(struct tty_struct *tty, unsigned int duration)
 		retval = tty->ops->break_ctl(tty, duration);
 	else {
 		/* Do the work ourselves */
-		if (!tty_write_lock(tty, 0))
+		if (tty_write_lock(tty, 0) < 0)
 			return -EINTR;
 		retval = tty->ops->break_ctl(tty, -1);
 		if (retval)
-- 
2.17.1

